specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,315 @@
+  /// Note that this position starts at 0.
+  [propget] HRESULT LineNumber([out, retval] UINT32* value);
+  
+  /// The column number of the source where the exception occurred.
+  /// In the JSON it is `exceptionDetail.columnNumber`.
+  /// Note that this position starts at 0.
+  [propget] HRESULT ColumnNumber([out, retval] UINT32* value);
+
+  /// The Name is the exception's class name.
+  /// In the JSON it is `exceptionDetail.exception.className`.
+  /// This is the empty string if the exception doesn't have a class name.
+  /// This can happen if the script throws a non-Error object such as `throw "abc";`
+  [propget] HRESULT Name([out, retval] LPWSTR* value);
+
+  /// The Message is the exception's message and potentially stack.
+  /// In the JSON it is exceptionDetail.exception.description.
+  /// This is the empty string if the exception doesn't have a description.
+  /// This can happen if the script throws a non-Error object such as throw "abc";.
+  [propget] HRESULT Message([out, retval] LPWSTR* value);
+
+  /// This will return all details of the exception as a JSON string.
+  /// In the case that script has thrown a non-Error object such as `throw "abc";`
+  /// or any other non-Error object, you can get object specific properties.
+  [propget] HRESULT ExceptionAsJSON([out, retval] LPWSTR* value);
+}
+
+/// This is the result for ExecuteScriptWithResult.
+[uuid(D2C59C5C-AD36-4CF4-87CF-2F5359F6D4CB), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptResult : IUnknown {
+
+  /// This property is true if ExecuteScriptWithResult successfully executed script with
+  /// no unhandled exceptions and the result is available in the ResultAsJson property
+  /// or via the TryGetResultAsString method.
+  /// If it is false then the script execution had an unhandled exception which you
+  /// can get via the Exception property.
+  [propget] HRESULT IsSuccess([out, retval] BOOL* value); 
+
+  /// If IsSuccess is true, then this property is the JSON representation of the result of the script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT ResultAsJson([out, retval] LPWSTR* jsonResult);
+
+  /// If IsSuccess is true and the result of script execution is a string, this method provides the value of the string result
+  /// and otherwise returns E_INVALIDARG.
+  HRESULT TryGetResultAsString([out, retval] LPWSTR* stringResult);
+
+  /// If IsSuccess is false, you can use this property to get the unhandled exception thrown by script execution
+  /// and otherwise returns E_INVALIDARG.
+  [propget] HRESULT Exception(
+      [out, retval] ICoreWebView2ExecuteScriptException** exception);
+}
+
+/// This is the callback for ExecuteScriptWithResult
+[uuid(CECDD25B-E6E8-4A4E-B890-BBF95932564F), object, pointer_default(unique)]
+interface ICoreWebView2ExecuteScriptWithResultCompletedHandler : IUnknown {
+
+  /// Provides the result of ExecuteScriptWithResult
+  HRESULT Invoke(
+      [in] HRESULT errorCode, 
```

Comment:
Consider putting the hresult into the `-Exception` type as a property named `ExtendedError`

Reply:
No change.

Discussion: This HRESULT is the errorCode for running the script as opposed to any errors that are thrown by the script. Errors thrown by the script are represented in the Result object but because they are JavaScript errors we don't have an underlying HRESULT.